src/main/java/com/internet/shop/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,47 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.OrderDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Order;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public Order create(Order order) {
+        Storage.addOrder(order);
+        return order;
+    }
+
+    @Override
+    public Optional<Order> getById(Long id) {
+        return getAll().stream()
+                .filter(order -> order.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public Order update(Order order) {
+        List<Order> orders = getAll();
+        for (int i = 0; i < orders.size(); i++) {
+            if (orders.get(i).getId().equals(order.getId())) {
+                orders.set(i, order);
+                return order;
+            }
+        }
+        throw new IllegalArgumentException("Storage doesn't contain order with id "
+                                           + order.getId());
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        return getAll().removeIf(order -> order.getId().equals(id));
```

Comment:
```suggestion
        return Storage.orders.removeIf(order -> order.getId().equals(id));
```
на даному рівні можна й напряму звертатись до Storage

Reply:
То что можно я понимаю, но чем это лучше? По моему оба варианта вполне допустимы, разве нет?

Reply:
так два варіанта допустимі, просто написала що так також можна)